### DISCUSSION TITLE
Behavior of `TerminateAsync`

### DISCUSSION THREAD

**Discussion by lilyjma** [Azure/azure-functions-durable-extension#2279] [Q&A]
I have this code:

```c# 
        [FunctionName("lilyDotnetDFTest_HttpStart")]
        public static async Task<HttpResponseMessage> HttpStart(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] HttpRequestMessage req,
            [DurableClient] IDurableOrchestrationClient starter,
            ILogger log)
        {
            // Function input comes from the request content.
            string instanceId = await starter.StartNewAsync("lilyDotnetDFTest", null);

            log.LogInformation($"Started orchestration with ID = '{instanceId}'.");

            await starter.TerminateAsync(instanceId, "testing termination");

            return starter.CreateCheckStatusResponse(req, instanceId);
        }
```

I went to the `statusQueryGetUri` url and saw that the runtime status is terminated. However, on Azure Portal, I still see the orchestration as running:
<img width="697" alt="image" src="https://user-images.githubusercontent.com/25314976/192597990-9515b691-4ed5-4159-8dec-b2d9cf565ae8.png">

Orchestration details show that the orchestration was terminated, but then started 
<img width="551" alt="image" src="https://user-images.githubusercontent.com/25314976/192598258-5ea50468-fe1b-42f6-b0c4-f23656f070f6.png">

@davidmrdavid explained that this is because `TerminateAsync` terminates _running_ orchestrations, not scheduled ones. What the code above does is schedule an orchestration, then immediately terminates it. But at that point, there's no orchestration running yet, so there's nothing to terminate. And so what was originally scheduled would still run. Is this the expected behavior? How would I cancel a scheduled orchestration then?

**Comment by cgillum** [comment]
@lilyjma I suspect you hit some kind of race condition where the "start" and "terminate" messages are racing with each other. I don't think it's very common for people to start and then terminate their orchestration in the same function, so I assume not many people run into this. It is odd, however, that the runtime status you get from `statusQueryGetUri` shows `Terminated` even though the orchestration appears to have run.

What happens if you put an `await Task.Delay(TimeSpan.FromSeconds(5));` between the start and terminate calls of your function?

**Reply by lilyjma** [reply]
Apologies for the late response. 

I did what you instructed and for the specific orchestration instance, I got `terminated` in `statusQueryGetUri`, runtime status as `running` in orchestration trace, and `awaited` (after `terminated`) in orchestration trace detail

<img width="591" alt="image" src="https://user-images.githubusercontent.com/25314976/196756651-07e9cbf0-6a4f-450a-b13b-d261bff0cd37.png">

**Reply by davidmrdavid** [reply]
I've added an item on our taskboard (in backlog) to track an investigation for this.
